shell/platform/android/io/flutter/plugin/common/StandardCodec.java

```diff
@@ -0,0 +1,334 @@
+package io.flutter.plugin.common;
+
+import java.io.ByteArrayOutputStream;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A {@link MessageCodec} using binary messages in Flutter standard encoding.
```

Comment:
No. The idea was that we could use any protocol, as long as at is understood by the standard codec for the Android FlutterChannel, iOS FlutterChannel, and Dart PlatformChannel triple of a given Flutter SDK release. There are obvious trade-offs involved in using an externally specified protocol (maintenance, coupling, performance, etc).

As an example, I currently use a single byte to distinguish between success and error result envelopes, and nobody outside my three StandardMessageCodecs need to know.